File: trunk/activeobject/src/synchronized_domain_name_registry.rb
require "src/domain_name_registry"

class SynchronizedDomainNameRegistry
  def initialize(domain_space)
    @domain_name_registry = DomainNameRegistry.new(domain_space)
    @mutex = Mutex.new
  end
  def buy_domain_name(domain_name)
    @mutex.synchronize {
      @domain_name_registry.buy_domain_name(domain_name)
    }
  end
end

File: trunk/activeobject/java/src/activeobject/SynchronizedDomainNameRegistry.java
package activeobject;


public class SynchronizedDomainNameRegistry {

	private final DomainNameRegistry domainNameRegistry;

	public SynchronizedDomainNameRegistry(String domainSpace) {
		domainNameRegistry = new DomainNameRegistry(domainSpace);
	}
	
	public synchronized PurchaseResult buyDomainName(String domainName)
	{
		return domainNameRegistry.buyDomainName(domainName);
	}
}
